sos/collector/clusters/ceph.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2022 Red Hat Inc., Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import json
+
+from sos.collector.clusters import Cluster
+
+
+class ceph(Cluster):
+    """
+    This cluster profile is for Ceph Storage clusters, and is primarily
+    built around Red Hat Ceph Storage 5. Nodes are enumerated via `cephadm`; if
+    your Ceph deployment uses cephadm but is not RHCS 5, this profile may work
+    as intended, but it is not currently guaranteed to do so. If you are using
+    such an environment and this profile does not work for you, please file a
+    bug report detailing what is failing.
+
+    By default, all nodes in the cluster will be returned for collection. This
+    may not be desireable, so users are encouraged to use the `labels` option
+    to specify a colon-delimited set of ceph node labels to restrict the list
+    of nodes to.
+
+    For example, using `-c ceph.labels=osd:mgr` will return only nodes labeled
+    with *either* `osd` or `mgr`.
```

Comment:
Why colon and not comma (that we use in other multi-value arguments, like `sos report -o qpid,logs,pulp`)?

Or is this syntax more convenient for ceph users or for cluster..?

I am ok with whatever works well / the best for the users (where I lack their preferences).

Reply:
This is a hold over from the structure of plugin options that was also applied to cluster options. Because plugin options are passed as  _values_ of the `-k`/`--plugin-option` option, we can't have commas in the values for those plugin options. Argparse instead uses the comma delimiter for additional values of `-k`. E.G. `sos report -k foo.bar=123,abc.opt=456` equates to `sos report -k foo.bar=123 -k abc.opt=456`. It's only internally (external to argparse) that we sort out that the string `foo.bar=123` is itself an option that needs further parsing.

We do the same colon delimitation for the OCP cluster for this same reason.

Reply:
I see, that makes sense. And worth to have the consistency within - among clusters.